content/300-guides/050-database/830-prismas-type-system.mdx

```diff
@@ -0,0 +1,152 @@
+
+### Native type mappings
+
+Sometimes you may need to use a more specific database type that is not one of the default type mappings for your Prisma type. For this purpose, Prisma provides [native type attributes](/concepts/components/prisma-schema/data-model#native-types-mapping) to refine the core scalar types. For example, in the `createdAt` field of your `Post` model above you may want to use a date-only column in your underlying PostgreSQL database, by using the `date` type instead of the default type mapping of `timestamp(3)`. To do this, add a `@db.Date` native type attribute to the `createdAt` field:
+
+```prisma file=schema.prisma
+model Post {
+  id        Int      @id
+  title     String
+  createdAt DateTime @db.Date
+}
+```
+
+## How to introspect database types
+
+When you introspect your database, Prisma will take the database type of each table column and represent it in your Prisma schema using the correct Prisma type for the corresponding model field. If the database type is not the default database type for that Prisma scalar type, Prisma will also add a native type attribute.
+
+As an example, take a `User` table in a PostgreSQL database, with:
+
+- an `id` column with a data type of `serial`
+- a `name` column with a data type of `text`
+- an `isActive` column with a data type of `boolean`
+
+You can create this with the following SQL command:
+
+```sql
+CREATE TABLE "public"."User" (
+  id serial PRIMARY KEY NOT NULL,
+  name text NOT NULL,
+  "isActive" boolean NOT NULL
+);
+```
+
+Introspect your database with the following command run from the root directory of your project:
+
+```terminal
+npx prisma db pull
+```
+
+You will get the following Prisma schema:
+
+```prisma file=schema.prisma
+model User {
+  id       Int     @id @default(autoincrement())
+  name     String
+  isActive Boolean
+}
+```
+
+The `id`, `name` and `isActive` columns in the database are mapped respectively to the `Int`, `String` and `Boolean` Prisma types respectively. These are the default database types for these Prisma types, so Prisma does not add any native type attributes.
+
+Now add a `createdAt` column to your database with a data type of `date` by running the following SQL command:
+
+```sql
+ALTER TABLE "public"."User"
+ADD COLUMN "createdAt" date NOT NULL;
+```
+
+Now introspect your database again:
+
+```terminal
+npx prisma db pull
+```
+
+Your Prisma schema now includes the new `createdAt` field with a Prisma type of `DateTime`. The `createdAt` field also has a `db.Date` native type attribute, because `date` is not the default type for the `DateTime` type:
```

Comment:
```suggestion
Your Prisma schema now includes the new `createdAt` field with a Prisma type of `DateTime`. The `createdAt` field also has a `@db.Date` native type attribute, because PostgreSQL's `date` is not the default type for the `DateTime` type:
```